src/Storages/VirtualColumnUtils.h

```diff
@@ -47,7 +49,7 @@ auto extractSingleValueFromBlock(const Block & block, const String & name)
 }
 
 NameSet getVirtualNamesForFileLikeStorage();
-VirtualColumnsDescription getVirtualsForFileLikeStorage(const ColumnsDescription & storage_columns);
+VirtualColumnsDescription getVirtualsForFileLikeStorage(const ColumnsDescription & storage_columns, Strings paths = {});
```

Comment:
BTW, why do we need more than 1 path here? I beleive we can find all keys from single path, in hive partitioning all paths should be the same (and if they are not, maybe it's not real hive partitioning and we don't need to do anything).